internal/controller/template_controller.go

```diff
@@ -359,6 +359,11 @@ func fillStatusWithProviders(template templateCommon, helmChart *chart.Chart) er
 func (r *TemplateReconciler) updateStatus(ctx context.Context, template templateCommon, validationError string) error {
 	status := template.GetCommonStatus()
 	status.ObservedGeneration = template.GetGeneration()
+
+	if validationError != "" && strings.Contains(validationError, "failed to get source:") && strings.Contains(validationError, "not found") {
```

Comment:
Please check whether the error is at least `apierrors.IsNotFound()`. I also suggest putting the check in the place where the actual `"github.com/fluxcd/source-controller/api/v1".HelmChart` is being retrieved, and check its conditions on that particular `"github.com/fluxcd/source-controller/api/v1".FetchFailedCondition` and its reason set to `SourceUnavailable` along with that check of the underlying error. That would be much better than checking for arbitrary errors that contain another arbitrary string that could change at any moment.